free.cabal

```diff
@@ -79,7 +79,7 @@ library
     profunctors          >= 4 && < 6,
     semigroupoids        >= 4 && < 6,
     transformers         >= 0.2.0   && < 0.6,
```

Comment:
This line is now redundant, correct? Let's remove it (and migrate the upper version bounds below, to where we declare a dependency on `transformers` within `if impl...`).

Reply:
It's not, other branch below doesn't have `transformers` dependency. It's either
- recent enough `transformers` or
- very recent `transformers-compat`

Reply:
OK. As said in https://github.com/ekmett/tagged/pull/44/files#r192603640, a brief comment explaining what this is doing would be appreciated.

Reply:
amended